backend/src/apiserver/storage/pipeline_store.go

```diff
@@ -82,16 +84,29 @@ type PipelineStore struct {
 
 // Runs two SQL queries in a transaction to return a list of matching pipelines, as well as their
 // total_size. The total_size does not reflect the page size.
-func (s *PipelineStore) ListPipelines(opts *list.Options) ([]*model.Pipeline, int, string, error) {
+func (s *PipelineStore) ListPipelines(filterContext *common.FilterContext, opts *list.Options) ([]*model.Pipeline, int, string, error) {
 	errorF := func(err error) ([]*model.Pipeline, int, string, error) {
 		return nil, 0, "", util.NewInternalServerError(err, "Failed to list pipelines: %v", err)
 	}
 
 	buildQuery := func(sqlBuilder sq.SelectBuilder) sq.SelectBuilder {
-		return opts.AddFilterToSelect(sqlBuilder).
-			From("pipelines").
-			LeftJoin("pipeline_versions ON pipelines.DefaultVersionId = pipeline_versions.UUID").
-			Where(sq.Eq{"pipelines.Status": model.PipelineReady})
+		query := opts.AddFilterToSelect(sqlBuilder).From("pipelines").
+			LeftJoin("pipeline_versions ON pipelines.DefaultVersionId = pipeline_versions.UUID")
+		if filterContext.ReferenceKey != nil && filterContext.ReferenceKey.Type == common.Namespace {
+			glog.Info("Using Namespace to filter the query")
+			query = query.Where(
+				sq.Or{
+					sq.Eq{"pipelines.Status": model.PipelineReady,
+						"pipelines.Namespace": filterContext.ReferenceKey.ID},
+					sq.Eq{"pipelines.Status": model.PipelineReady,
+						"pipelines.Namespace": ""}},
```

Comment:
Is List returning both namespaced and non-namespaced pipelines?
I thought we had agreed not to return both: https://github.com/kubeflow/pipelines/issues/4197#issuecomment-730571962

Reply:
@yanniszark Yeah that makes sense, I'll create a small switch in the UI to return either "Shared" or "Namespaced" pipelines only.

Reply:
Commented outside, I think we need to resolve to a conclusion, both options have pros and cons